src/main/java/org/testng/xml/internal/TestNamesHelper.java

```diff
@@ -0,0 +1,97 @@
+package org.testng.xml.internal;
+
+import java.util.Iterator;
+import java.util.List;
+import org.testng.TestNGException;
+import org.testng.collections.Lists;
+import org.testng.xml.XmlSuite;
+import org.testng.xml.XmlTest;
+
+/**
+ * A helper class to work with "-testnames"
+ *
+ */
+public final class TestNamesHelper {
```

Comment:
I propose something like `TestNameMatcher`

Reply:
Which makes this class close to what it does, great, I like this name.

Reply:
@krmahadevan  how do you think about the `TestNameMatcher`? If you are fine with it, I will nail it down. :)

Reply:
Yeah. The name is pretty apt. I am fine with it as well.